contracts/InstadappTarget.sol

```diff
@@ -0,0 +1,95 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IConnext} from "@connext/interfaces/core/IConnext.sol";
+import {IXReceiver} from "@connext/interfaces/core/IXReceiver.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {InstadappAdapter} from "./InstadappAdapter.sol";
+
+/// @title InstadappTarget
+/// @author Connext
+/// @notice You can use this contract for cross-chain casting via dsa address
+/// @dev This contract is used to receive funds from Connext
+/// and forward them to Instadapp DSA via authCast function, In case of failure,
+/// funds are forwarded to fallback address defined by the user under callData.
+/// @custom:experimental This is an experimental contract.
+contract InstadappTarget is IXReceiver, InstadappAdapter {
+  using SafeERC20 for IERC20;
+  /// Storage
+  /// @dev This is the address of the Connext contract.
+  IConnext public connext;
+
+  /// Events
+  /// @dev This event is emitted when the authCast function is called.
+  event AuthCast(bytes32 transferId, address dsaAddress, address auth, bool success, bytes returnedData);
```

Comment:
I think we can make `transferId`, `dsaAddress` as indexed param?

Reply:
make sense!!